test/functional/fixtures/before-test.js

```diff
@@ -0,0 +1,146 @@
+var SauceLabsConnector     = require('saucelabs-connector');
+var Promise                = require('pinkie');
+var caller                 = require('caller');
+var path                   = require('path');
+var os                     = require('os');
+var Site                   = require('../site');
+var getErrorByBrowsersInfo = require('../helpers/report-helper.js');
+var config                 = require('../config.js');
+
+var testCafe     = null;
+var browsersInfo = null;
+
+var slConnector = null;
+var slBrowsers  = null;
+
+var SAUCELABS_SETTINGS = {
+    username:  process.env.SAUCE_USERNAME,
+    accessKey: process.env.SAUCE_ACCESS_KEY,
+    build:     process.env.TRAVIS_JOB_ID || '',
+    tags:      [process.env.TRAVIS_BRANCH || 'master'],
+    name:      'testcafe functional tests'
+};
+
+before(function () {
+    var hostname = process.env.TRAVIS ? os.hostname() : config.testCafe.hostname;
+
+    return createTestCafe(hostname, config.testCafe.port1, config.testCafe.port2)
+        .then(function (tc) {
+            var startPromise        = null;
+            var openBrowserPromises = null;
+
+            testCafe = tc;
+
+            browsersInfo = config.browsers
+                .filter(function (browser) {
+                    return SAUCELABS_SETTINGS || !browser.remoteOnly;
+                })
+                .map(function (settings) {
+                    return {
+                        settings:   settings,
+                        connection: tc.createBrowserConnection()
+                    };
+                });
+
+            Site.create(config.site.port1, config.site.port2, config.site.viewsPath);
+
+            if (process.env.TRAVIS) {
+                slConnector = new SauceLabsConnector(SAUCELABS_SETTINGS.username, SAUCELABS_SETTINGS.accessKey);
+
+                startPromise = slConnector
+                    .connect()
+                    .then(function () {
+                        openBrowserPromises = browsersInfo.map(function (browserInfo) {
+                            return slConnector.startBrowser(browserInfo.settings, browserInfo.connection.url,
+                                SAUCELABS_SETTINGS.jobName);
+                        });
+
+                        return Promise.all(openBrowserPromises);
+                    })
+                    .then(function (browsers) {
+                        slBrowsers = browsers;
+                    });
+            }
+            else {
+                openBrowserPromises = browsersInfo.map(function (browserInfo) {
+                    return browserNatives.getBrowserInfo(browserInfo.settings.alias)
+                        .then(function (browser) {
+                            return browserNatives.open(browser, browserInfo.connection.url);
+                        });
+                });
+
+                startPromise = Promise.all(openBrowserPromises);
+            }
+
+            return startPromise;

```

Comment:
it's not understandable 'start what?' from this var name.
I'd prefer to get rid of it:
```js
if(process.env.TRAVIS) {
    ...
    return slConnector.connect().then()....
}

openBrowserPromises = ...
return Promise.all(openBrowserPromises);
```